test/unit/eHealth.locations/services/selected-location-factory.js

```diff
@@ -457,4 +452,39 @@ describe('Service: SelectedLocationFactory', function () {
       expect(location.levels.length).toBe(3);
     });
   });
+
+  describe('common behaviour', function () {
```

Comment:
This was not as straightforward as expected, so I decided against it. `with locations for Mali` has nested contexts that each setup `locations` differently. In order to refactor the this test suite properly the contexts would probably have to be completely reorganized. Did not feel like it is worth it.